docs/move-out-of-stock-products-to-the-end-of-a-collection/script.liquid

```diff
@@ -167,36 +192,24 @@
 
   {% assign moves = array %}
 
+  {% assign current_product_ids_sorted = collection.products | map: "admin_graphql_api_id" %}
```

Comment:
> I added the ability to short-circuit a collection reorder IF the new sort order exactly matches the current order (more accurately, I do this for each product/position). I am using the product ids mapped from collection.products for this, as they are returned in their current sort order. I do not know if there are any memory issues with this approach for collections with a lot of products.

I think this will be a problem for the kind of stores that'll benefit from this refactor - i.e. stores with many collections. Pulling `{{ collection.products | map: "anything" }}` will mean downloading the complete REST representation for every product in the collection, which (for stores running this on all collections) will mean pulling down a ton of redundant product data.

Can you replace this bit with a paginating GraphQL query, pulling out product IDs that way?

Reply:
Yep, will do